src/BrazeKit-dev.js

```diff
@@ -66,32 +66,92 @@ var constructor = function () {
         dob: 'setDateOfBirth',
     };
 
+    var bundleProductsWithEvent = false;
+
+    // A purchase event can either log a single event with all products
+    // or multiple purchase events (one per product)
     function logPurchaseEvent(event) {
         var reportEvent = false;
-        if (event.ProductAction.ProductList) {
-            event.ProductAction.ProductList.forEach(function (product) {
-                if (product.Attributes == null) {
-                    product.Attributes = {};
+
+        if (bundleProductsWithEvent) {
+            reportEvent = logSinglePurchaseEventWithProducts(event);
+        } else {
+            reportEvent = logPurchaseEventPerProduct(event);
+        }
+        return reportEvent === true;
+    }
+
+    function logSinglePurchaseEventWithProducts(event) {
+        var quantity = 1;
+        var eventAttributes = mergeObjects(event.EventAttributes, {
+            products: [],
+        });
```

Comment:
I see you're checking two different child attributes of `event.ProductAction` below, but you're not checking if `event.ProductAction` exists anywhere. Not sure if I'm being too nipticky but I think you should check here and return false or throw an error somewhere.

I think this can also be done in `logPurchaseEvent` if that's easier/cleaner.

Reply:
Good call.  I actually caught this in the nonPurchase event logging function when trying to fire impression/promotion events.  i'll fix that here too

Reply:
It's actually the case that on purchase events, ProductAction will always exist, but I'll include a return false and put a comment there.